File: charts/code/style.R
# load packages ####
  library(here)
  library(showtext)
  library(tidyverse)
  library(cowplot)

# assigning colors to objects ####
  
  # I got these hex codes from the GW Web Accessible Color Guidelines

  # core primary colors
    GWblue <- "#033C5A"
    GWbuff <- "#A69362"
    lightblue <- "#0073AA"
    lightyellow <- "#F8E08E"
    
  # secondary colors
    darkyellow <- "#FFC72C"
    brown <- "#A75523"
    darkgreen <- "#008364"
    lightgreen <- "#78BE20"
    red <- "#C9102F"
    
    RSCgray <- "#E0E0E0"
      # saw we used this gray a few times for grid lines, so added it as a color!
    fill <- "#B2DDF4"
      # this is the color used in the striping on patterned charts
    
# logo ####
   logo <- image_read(here("charts", "style", "gw_ci_rsc_2cs_pos.png"))
    
# font ####
  font_add("avenir_lt_pro", here("charts", "style", "a-avenir-next-lt-pro.otf"))
  showtext_auto()
    # according to documentation, showtext_auto needed for this to work properly 
  
# custom theme for RSC Reg Stats plots
  theme_RSC <- theme_minimal() +
    theme(
      plot.title = element_text(hjust = 0.5, vjust = 0, margin = margin(b = 10, unit = "pt"), size = 20),
      legend.position = "none",
      panel.grid.major.x = element_blank(),
      axis.text.x = element_text(angle = 65, hjust = 0.5, vjust = 1, size = 12),
      axis.title.x = element_text(size = 12, angle = 0, vjust = 0.5, hjust = 0.5, margin = margin(t = 10)),
      axis.ticks.x = element_line(color = RSCgray),
      axis.text.y = element_text(size = 12),
      axis.title.y = element_text(size = 12, angle = 90, vjust = 0.5, hjust = 0.5, margin = margin(r = 10)),
      panel.grid.major.y = element_line(color = RSCgray, linetype = "solid", linewidth = 0.50),
      panel.grid.minor = element_blank(),
      text = element_text(family = "avenir_lt_pro"),
      plot.caption = element_text(hjust = 1, vjust = 0, margin = margin(t = 10, l = 6, unit = "pt"), size = 11),
      plot.margin = margin(50, 50, 50, 50)
    )
  
  # thank you to Yaxin for developing this custom theme!
  

File: dashboard-two/utilis/style.py
import pandas as pd
from pyprojroot import here
from plotnine import *
from PIL import Image
import matplotlib.font_manager as fm

# colors
GW_COLORS = {
    "GWblue": "#033C5A",
    "GWbuff": "#A69362",
    "lightblue": "#0073AA",
    "lightyellow": "#F8E08E",
    # Secondary colors
    "darkyellow": "#FFC72C",
    "brown": "#A75523",
    "darkgreen": "#008364",
    "lightgreen": "#78BE20",
    "red": "#C9102F",
    "RSCgray": "#E0E0E0",
    "RSCdarkgray": "#bdbdbd",
    "fill": "#B2DDF4"
}

# logo
logo_path = here("dashboard-two/utilis/style/gw_ci_rsc_2cs_pos.png")
logo = Image.open(logo_path)

# font
font_path = str(here("dashboard-two/utilis/style/a-avenir-next-lt-pro.otf"))
fm.fontManager.addfont(font_path)
custom_font = fm.FontProperties(fname=font_path).get_name()

theme_RSC = theme_minimal(base_family=custom_font) + \
    theme(
        plot_title=element_text(ha='center', va='bottom', margin={'b': 25}, size=20),
        legend_position="none",
        panel_grid_major_x=element_blank(),
        axis_text_x=element_text(angle=65, ha='right', va='top', size=12),
        axis_title_x=element_text(size=12, margin={'t': 12}),
        axis_ticks_major_x=element_line(color=GW_COLORS["RSCgray"]),
        axis_text_y=element_text(size=12),
        axis_title_y=element_text(size=12, angle=90, margin={'r': 12}),
        panel_grid_major_y=element_line(color=GW_COLORS["RSCgray"], linetype="solid", size=0.5),
        panel_grid_minor=element_blank(),
        plot_caption=element_text(ha='right', va='bottom', margin={'t': 10}, size=11),
        plot_margin=50 # Plotnine uses a single value or a dictionary for margins
    )